### PR TITLE
feat(assistant builder): scaffolding and basic layout

### PR DESCRIPTION
**Pull request by fontanierh**
Just the layout, 0 functionality for now.
Only accessible in Dev or for the Dust workspace.

In a follow up PR, i'll:
- move the input box to a sparkle component: being done [here](https://github.com/dust-tt/dust/pull/1305)
- ~~will add the "chevron up/down" icon to sparkle~~ Added by Duncid [here](https://github.com/dust-tt/dust/pull/1297)
- ~~will create a new component for the "timeframe selector drop down"~~ Added by @Duncid  [here](https://github.com/dust-tt/dust/pull/1297) ✨ 

Screenshot 2023-09-07 at 14 56 50
Screenshot 2023-09-07 at 14 57 13

### CODE REVIEW

File: front/components/sparkle/navigation.tsx [hunk 1]
```diff
@@ -139,6 +144,17 @@ export const subNavigationAdmin = ({
       subMenuLabel: current === "developers" ? subMenuLabel : undefined,
       subMenu: current === "developers" ? subMenu : undefined,
     });
+    if (isDevelopmentOrDustWorkspace(owner)) {
+      nav.push({
+        id: "assistants",
+        label: "Assistants Manager",
```
Comment (PopDaph) [inline front/components/sparkle/navigation.tsx:150]: NIT: Assistant Builder?

**Review by PopDaph — APPROVED** [review]

File: front/components/sparkle/navigation.tsx (hunk 1 above)
Comment (fontanierh) [reply front/components/sparkle/navigation.tsx:150]: took it from the design but agreed builder is more consistent

File: front/pages/w/[wId]/builder/assistants/index.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
+import {
+  PageHeader,
+  PlusIcon,
+  RobotStrokeIcon,
```
Comment (spolu) [inline front/pages/w/[wId]/builder/assistants/index.tsx:4]: Should be non stroke

File: front/pages/w/[wId]/builder/assistants/new.tsx [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+      titleChildren={
+        <AppLayoutSimpleCloseTitle
+          title="Design your custom Assistant"
+          onClose={() => {
+            void router.push(`/w/${owner.sId}/builder/assistants`);
+          }}
+        />
+      }
+    >
+      <div className="mt-8 flex flex-col space-y-8 pb-8">
```
Comment (spolu) [inline front/pages/w/[wId]/builder/assistants/new.tsx:84]: You should not need that mt-8?

File: front/pages/w/[wId]/builder/assistants/new.tsx [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+        <PageHeader
+          title="Assistant Editor"
+          icon={RobotStrokeIcon}
+          description="Make and maintain your customized assistants."
+        />
+        <div className="mt-8 flex flex-row items-start">
```
Comment (spolu) [inline front/pages/w/[wId]/builder/assistants/new.tsx:90]: This one either if you have a space-y-8 above?

**Pull request merged**